entities/NetworkComponent.ts

```diff
@@ -0,0 +1,84 @@
+import { Api } from "@cennznet/api";
+import { SubmittableExtrinsic } from "@cennznet/api/types";
+import { defineComponent } from "./entityHelpers";
+import { u8aToHex } from "@polkadot/util";
+
+const networkStore = {} as any;
+
+const CENNZnetWSS = {
+  CENNZnet_Azalea: "wss://cennznet.unfrastructure.io/public/wss",
+  CENNZnet_Nikau: "wss://nikau.centrality.me/public/ws",
+  CENNZnet_Rata: "wss://kong2.centrality.me/public/rata/ws",
+  Mock: "",
+};
+
+const NetworkComponent = defineComponent<
+  {},
+  {
+    apiConnector: () => Promise<Api>;
+    createTransaction: (input: {
+      address: string;
+      extrinsic: SubmittableExtrinsic<any, any>;
+    }) => Promise<{
+      expectedSigningAddress: { address: string };
+      signerPayload: any;
+      id: string;
+    }>;
+  }
+>({
+  __component: "Network",
+  cacheKey() {
+    return this.__network;
+  },
+  capability: {
+    async apiConnector() {
+      if (!networkStore[this.__network] && CENNZnetWSS[this.__network]) {
+        networkStore[this.__network] = Api.create({
+          provider: CENNZnetWSS[this.__network],
+        });
+      }
+      if (!networkStore[this.__network]) {
+        throw new Error(`${this.__network} is currently unsupported`);
+      }
+      return networkStore[this.__network];
+    },
+    async createTransaction({ address, extrinsic }) {
+      const api = await this.apiConnector();
```

Comment:
should this logic be enclosed in a try catch block?

Reply:
should only use try catch when you actually want to do something with the error, otherwise your being defensive in such a way that's more frustrating.